relay-server/src/actors/store.rs

```diff
@@ -1100,13 +1104,20 @@ impl Message for KafkaMessage<'_> {
             Self::Attachment(message) => message.event_id.0,
             Self::AttachmentChunk(message) => message.event_id.0,
             Self::UserReport(message) => message.event_id.0,
-            Self::Session(_message) => Uuid::nil(), // Explicit random partitioning for sessions
-            Self::Metric { .. } => Uuid::nil(),     // TODO(ja): Determine a partitioning key
-            Self::Profile(_message) => Uuid::nil(),
             Self::ReplayEvent(message) => message.replay_id.0,
-            Self::ReplayRecordingNotChunked(_message) => Uuid::nil(), // Ensure random partitioning.
-            Self::CheckIn(_message) => Uuid::nil(),
-            Self::Span(_) => Uuid::nil(), // random partitioning
+
+            // Monitor check-ins use the hinted UUID passed through from the Envelope.
+            //
+            // XXX(epurkhiser): In the future it would be better if all KafkaMessage's would
+            // recieve the routing_key_hint form their envelopes.
+            Self::CheckIn(message) => message.routing_key_hint.unwrap_or_else(Uuid::nil),
+
+            // Random partitioning
+            Self::Session(_)
+            | Self::Metric { .. }
```

Comment:
Please keep the TODO on metrics, and make it a separate match arm, I still want to address this eventually.

Reply:
Ah sorry I meant to bring it back, my fault